src/main/java/com/epam/aidial/cfg/domain/mapper/RoleLimitMapper.java

```diff
@@ -40,54 +42,47 @@ default Map<String, CoreLimit> mapLimits(List<RoleLimit> roleLimits, @Context Co
                 .map(roleLimit -> {
                     Deployment deployment = deploymentsByName.get(roleLimit.getDeploymentName());
                     CoreLimit mappedLimit = mapLimit(roleLimit.getLimit(), deployment);
-                    return mappedLimit != null ? new AbstractMap.SimpleEntry<>(roleLimit.getDeploymentName(), mappedLimit) : null;
+                    return new AbstractMap.SimpleEntry<>(roleLimit.getDeploymentName(), mappedLimit);
                 })
-                .filter(Objects::nonNull)
+                .filter(entry -> isLimited(entry.getValue()))
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
     }
 
     private CoreLimit mapLimit(Limit limit, Deployment deployment) {
         if (limit == null || deployment == null) {
-            return null;
+            log.warn("Limit or deployment is null. Limit: {}, deployment: {}", limit, deployment);
+            throw new IllegalStateException("Limit or deployment is null");
```

Comment:
IllegalArgumentException is more suitable here since we validate incoming parameters, not the state of object